feat: 프로젝트 생성

prk4224 opened the pull request:
일단 프로젝트만 생성 해놓고 bliud sync 만 되는 상태, 
빌드를 하기위해서 아래 의존성이 필요 
```kotlin
    id 'com.android.application' version '7.3.0'
    id 'com.android.library' version '7.3.0'
    id 'org.jetbrains.kotlin.android' version "${ktlint_version}"
```
#### 빌드까지 하려고 했지만 프로제트 구조가 이상해질까봐 같이 상의해서 하는게 좋을듯 합니다.

### Reference
- [Notion Ticket](https://www.notion.so/yapp-workspace/Android-adbd8527f4554a9aafe405a1f0764a38?pvs=4)

### 참고 사항

P1: 꼭 반영해주세요 (Request changes)
P2: 적극적으로 고려해주세요 (Request changes)
P3: 웬만하면 반영해 주세요 (Comment)
P4: 반영해도 좋고 넘어가도 좋습니다 (Approve)
P5: 그냥 사소한 의견입니다 (Approve)

prk4224 committed:
feat: gredle.properties 생성

prk4224 committed:
feat: App module 추가

ckrudals commented on app/build.gradle:
```diff
@@ -0,0 +1,44 @@
+
+android {
+    namespace 'com.jaehong.app'
+    compileSdk 32
```
p2 ) 저희 프로젝트는 멀티모듈로 구성하기로 논의가 된 상태니 compile, target sdk 버전은 app 단에서 상수로 지정하는 것이 좋을 것 같습니다.

prk4224 replied:
app 단에서 상수로 지정한다는 의미가 정확히 어떤 의미인지는 모르겠으나,(ext 로 지정한다는 뜻?) 그 정도로 빌드에 대한 변수? 관리? 를 한다고 가정하면 build module 을 따로 만들어서 bulid 만 따로 관리하는 형태로 만드는 것이 효과적일거 같아요.

islandparadise14 replied:
혹시 말씀하시는 상수화가 kts 적용했을때 코틀린으로 상수화 하는거 말씀하시는걸까요?

comforest replied:
root의 gradle.properties 을 사용할 수도 있어요

ckrudals replied:
저는 개인적으로 kotlin DSL 를 사용하는 것을 선호하긴 합니다. 어떤 것을 사용할 지 의논이 필요해 보여요

ckrudals commented on app/build.gradle:
```diff
@@ -0,0 +1,44 @@
+    compileSdk 32
+
+    defaultConfig {
+        applicationId "com.jaehong.app"
```
q) 패키지명은 임시로 지정하신거죠?

prk4224 replied:
일단 app module 은 필요할것으로 보여 추후 수정을 염두해 두고 있긴해요 !

ckrudals commented on app/build.gradle:
```diff
@@ -0,0 +1,44 @@
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
+        }
```
p2) 궁금한 것이 있는데, 처음 세팅하면 buildType이 release 밖에 생성이 안되나요? 현재 저희가 필요한 것은 `debug` type 인 것 같습니다.

prk4224 replied:
..? 전 모듈을 phone& tablet 으로 생성해서 자동 생성된 코드인거 같은데, 어떤 부분이 궁금한지 이해를 못했어요

eunjjungg replied:
저도 잘은 모르겠지만 프로젝트 생성하면 release만 생기는것 같긴 하더라구요~ 저도 그래서 항상 debug {} 추가했던 것 같습니다.. 😄

islandparadise14 replied:
저도 프로젝트 생성할때마다 default로 release 만 생성됐었어요
debug 바로 추가해서 반영해도 될것 같네요

ckrudals replied:
오.. 처음알았네요, 새로 프로젝트를 만든지 너무 오래되서;; 설명 감사합니다

ckrudals commented on app/src/main/res/values/strings.xml:
```diff
@@ -0,0 +1,3 @@
+<resources>
+    <string name="app_name">app</string>
```
q) 모듈 이름 을 app 으로 지정하신 이유가 따로 있으실까요? app 으로 이름을 지정할 때는 보통 2가지 인 것 같습니다.
- Contaner(main) 로 사용
- 상위 레벨에서 사용될 코드의 집합 (hilt)

prk4224 replied:
처음에 결정한 부분이 bol 이라는 모듈이였는데, 너무 추상적인 내용인거 같아서 임시적으로 app이라고 지었어요. (그래도 App module이라고 정의하면 어떤 역할을 하는지 알 수 있을테니?) 여튼 추후 수정할수도 있다고 생각해서 이렇게 지었어요 ~

ckrudals replied:
> 처음에 결정한 부분이 bol 이라는 모듈이였는데, 너무 추상적인 내용인거 같아서 임시적으로 app이라고 지었어요. (그래도 App module이라고 정의하면 어떤 역할을 하는지 알 수 있을테니?) 여튼 추후 수정할수도 있다고 생각해서 이렇게 지었어요 ~

네! 추후 논의를 톹해 정하는게 좋을 것 같습니다.

ckrudals commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -1,5 +1,6 @@
-zipStoreBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
 zipStorePath=wrapper/dists
+zipStoreBase=GRADLE_USER_HOME
```
p1) 불필요환 변경 같습니다.

prk4224 replied:
저게 정확히 어떤 의미인지 모르겠는데, 설명 한번 해주세요..😵

eunjjungg replied:
이거 뭔가 제가 생각하기엔.. 재홍님 안드로이드 스튜디오에서 열면서 자동으로 변경된거 같아요 🧐 아니면 의존성 추가하면서 자동으로 만들어진거 같아서 다른 사람이 협업할 때 오류나는 거 아니면 그냥 냅둬도 괜찮을거 같습니다..!!

prk4224 replied:
오호 그렇군..아 그리고 스튜디오 버전도 다 맞춰면 좋을 듯해요 !

ipf-kmcha replied:
제 추측으로는 gradle 다운받아진 경로를 뜻 하는 것 같습니다. 추후 `file change` 가 방대해지면 뜻하지 않는 변경사항이 올라가는 것을 사전 방지하기 위해 커멘트 달았어요 ㅎㅎ

ckrudals replied:
제 추측으로는 gradle 다운받아진 경로를 뜻 하는 것 같습니다.
추후 file change 가 방대해지면 뜻하지 않는 변경사항이 올라가는 것을 사전 방지하기 위해 커멘트 달았어요 ㅎㅎ

prk4224 replied:
수정 완료 3c8e01d7ce032a1141454840f70b3f0787de1f70

ckrudals reviewed (commented):
작업하시느라 고생하셨습니다. 👍 궁금한 내용은 `q)` 로 질문드렸어요 !

ckrudals reviewed (changes requested):
작업하시느라 고생하셨습니다. 👍 궁금한 내용은 `q)` 로 질문드렸어요 !

prk4224 committed:
Refactor: repositores google, mavenCentral -> settrings.gradle 이동

이유는 아직 모르겠으나, project bulid.gradle 에서 Sync 가 안됨. 검색해봤는데, 특정 버전부터는  project bulid.gradle repositores 를 사용하지  않는 것으로 추측

prk4224 committed:
Refactor: compileSdk Version Update

core,appcompat 등 라이브러리 버전이 compileSdk 33 에서 동작 가능
(Fyi: 다른 라이브러리 버전을 낮출지, compileSdk 버전을 올릴지 상의해보면 좋을듯합니다.)

prk4224 committed:
Refactor: GRADLE_USER_HOME 제거

ckrudals commented:
> core,appcompat 등 라이브러리 버전이 compileSdk 33 에서 동작 가능
(Fyi: 다른 라이브러리 버전을 낮출지, compileSdk 버전을 올릴지 상의해보면 좋을듯합니다.)

라이브러리 버전 문제로 sdk 버전을 올려야 하는 상황일 때 저 같은 경우면 위험도 측면에서 라이브러리 버전을 낮추는 것을 선호합니다. 
(compile sdk 버전이 올렸을 때의 위험도 > 라이브러리 버전을 낮출 때의 위험도)
위와 같은 경우면 라이브러리의 특정 기능을 사용할 때 고민하는데 지금같은 상황에서는 고려하지 않아도 될 것 같아요.
